code-challenges.test.js

```diff
@@ -15,42 +15,165 @@
 
 // a) Create a test with expect statements using the variable provided.
 
-const hand1 = [5, 5, 5, 3, 3]
-// Expected output: true
-const hand2 = [5, 5, 3, 3, 4]
-// Expected output: false
-const hand3 = [5, 5, 5, 5, 4]
-// Expected output: false
-
+describe ("fullHouse", () => {
+    it ("Create a function that takes in an array of 5 numbers and determines whether or not the array is a “full house”. A full house is exactly one pair and one three of a kind", () => {
+
+       const hand1 = [5, 5, 5, 3, 3]
+       // Expected output: true
+       const hand2 = [5, 5, 3, 3, 4]
+       // Expected output: false
+       const hand3 = [5, 5, 5, 5, 4]
+       // Expected output: false
+        
+        expect (fullHouse(hand1)).toEqual(true)
+        expect (fullHouse(hand2)).toEqual(false)
+        expect (fullHouse(hand3)).toEqual(false)
+    })
+})
 
 // b) Create the function that makes the test pass.
+
+// Create a function call theHouseIsfull
+// theHouseIsfull will take one parameter which will be an array
+// Use .sort to arrange from lowest to highest
+// Use the .slice to cut array and slice instances of a full house
+// Now this is the part of my code where I realized that I will have to create every instance of a full house
+// I started reseaching other functions to simplify the process and I came across a code that worked successfully. 
+
+// The function I started to create before my research...
+function theHouseIsfull (array) {
+
+    let sortArray = array.sort((a,b)=>a-b)
+
+    sliceHouse = sortArray.slice()
+
+    spliceHouse = sliceHouse.splice(i, 1)[0]
+      if(sliceHouse.includes(spliceHouse)) {
+
+    sliceHouse.splice(sliceHouse.indexOf(spliceHouse), 1)
+      if(sliceHouse.includes(spliceHouse)) {
+     return true
+ }
+ }else {
+         i++
+ }
+       
+        return false
+ }
+
+ console.log(theHouseIsfull(hand1))
+// true
+ console.log(theHouseIsfull(hand2))
+// false
+// My issue with the code I started... gives a true for (hand3)
+console.log(theHouseIsfull(hand3))
+// true
+
+// The code that I found and built on works perfect
+// The code create all instances of a full house
+// This problem was challenging for me because of the level of complexity in creating instances for all possiblies of a full house
+
+const fullHouse = (array) => {
+    
+       let sortArray = array.sort((a,b)=>a-b)
+    
+       if (sortArray[0] === sortArray[1] && sortArray[0] === sortArray[2] && sortArray[3] === sortArray[4]){
+         return true
+       }else if (sortArray[0] === sortArray[1] && sortArray[2] === sortArray[3] && sortArray[2] === sortArray[4])
+       {return true
+       }else
+       return false
+     }
+
+    console.log(fullHouse(hand1))
+    // true
+    console.log(fullHouse(hand2))
+    // false
+    console.log(fullHouse(hand3))
+    // false
+    
```

Comment:
Excellent work, Ricky! I like that you showed both attempts!  Way to work through the problem!